Testscripts/Linux/failsafe.sh

```diff
@@ -0,0 +1,384 @@
+$_self [vdev|with <net1[ net2 ...]>|preferred|fallback|help] [MAC]
+
+${BOLD}${YELLOW}$_self [vdev] <[MAC1[ MAC2 ...]]>${NORMAL}
+    Generates a vdev parameter for each VF-NetVSC pair found on the system.
+    If MAC is specified, restrict results to those matching such MAC address.
+    If the VF is not currently plugged-in and no bonding has been configured,
+    then MAC must be provided or using a specific NetVSC device using the
+    ${BOLD}with${NORMAL} command instead.
+
+    You can use it as follows:
+        ${BOLD}path/to/app \$(${BLUE}$self${BLACK})${NORMAL}
+
+${BOLD}${YELLOW}$_self with <net1[ net2 ...]>${NORMAL}
+    Use the specified netdevices as fallback for fail-safe pairs.
+    One fail-safe instance will be created per netdevice.
+    Each such netdevice will seek a VF matching its MAC address, be it currently
+    plugged-in or not.
+
+    You can use it as follows:
+        ${BOLD}path/to/app \$(${BLUE}$self ${RED}with eth1 eth2${BLACK})${NORMAL}
+
+${BOLD}${YELLOW}$_self preferred [MAC]${NORMAL}
+    Generates a device definition for the preferred device.
+    The preferred device is detected following those rules:
+        * Detect any PCI device currently plugged in that could be part of a VF-NetVSC pair.
+          This PCI device has to share a MAC address with another non-PCI device having
+          the class_id \"${RED}$netvsc_id${NORMAL}\".
+          If a MAC address is specified, restrict results to those using such MAC address.
+        * If no such device is detected and if a MAC address is specified, any
+          other PCI device using this MAC address is then used instead. This allows
+          forcing the use of arbitrary PCI device.
+
+    This option is usually used within the ${BOLD}exec${NORMAL} device type of the fail-safe.
+
+${BOLD}${YELLOW}$_self fallback [MAC]${NORMAL}
+    Generates a device definition for the fallback device.
+    The fallback device is a TAP device capturing the traffic from a NetVSC device.
+    It is detected using the following rules:
+        * Detect any VF-NetVSC pairs by listing all PCI device sharing a MAC address with
+          another device having the class_id \"${RED}$netvsc_id${NORMAL}\".
+        * If the VF is not currently plugged-in, detect any NetVSC device being
+          part of a bonding configuration.
+        * If a MAC address is specified, restrict results to those using such MAC address.
+        * If no such NetVSC device could be detected and a MAC address is specified, generate
+          a dummy TAP PMD command, then detect any possible non-PCI device on the system using
+          this MAC address, and use it as a remote.
+
+    This option is usually used within the ${BOLD}exec${NORMAL} device type of the fail-safe.
+"
+
+# printf, but to stderr
+err_printf () {
+    (>&2 printf %s "${BOLD}${RED}" && printf "$@" && printf %s "${NORMAL}")
+}
+
+# Print the MAC of a netdev
+# $1: netdev name
+get_mac () {
```

Comment:
@abhimarathe , @yuxisun1217 I see those functions are good candidates in dpdk utils lib file. Thought?

Reply:
So sorry I forgot to mention that this script is downloaded from https://gallery.technet.microsoft.com/Azure-DPDK-Failsafe-PMD-c34b1a4b . I just awared that maybe I should change to download it from a url instead of putting it in LISAv2 directly. What do you think about that? @juhlee-microsoft

Reply:
I just remove this file from LISAv2 and change to download it from url. The url is stored as FAILSAFE_SH_URL in ReplaceableTestParameters.xml.

Reply:
If this is the case, please add **the comment** line saying the source of download link and the license file type - MIT. We should include them legally.

Reply:
Hi @juhlee-microsoft ,
I just removed file from LISAv2 and change to download it from the URL. Do I still need to add the comment in dpdkPmdsBasicCheck.sh? Thanks!